intelmq/bots/parsers/spamhaus/parser_cert.py

```diff
@@ -47,7 +48,9 @@ def process(self):
             event.change("feed.url", event["feed.url"].split("key=")[0])
 
             event.add('source.ip', row_splitted[0])
-            event.add('source.asn', row_splitted[1].replace('AS', ''))
+            source_asn = row_splitted[1].replace('AS', '')
+            if source_asn != '?':
```

Comment:
is '?' the only option which the field may have instead of a numeric value?
What about if shadowserver renames the value to '???'  ?

Lot's of question marks ;-)

Reply:
It's not shadowserver, it's spamhaus.

We received a report with 'AS?', so I handle that now. I don't know what they sent tomorrow, probably they use an empty field? So I adapted to code to what we know they sent in the past. I try not to guess too much about the format.

Reply:
we always will have to adapt.